src/main/java/uk/gov/pay/publicauth/util/ApplicationStartupDependentResourceChecker.java

```diff
@@ -3,42 +3,30 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import java.sql.SQLException;
+import java.time.Duration;
+import java.util.function.Consumer;
 
-public class ApplicationStartupDependentResourceChecker {
+class ApplicationStartupDependentResourceChecker {
 
     private static final int PROGRESSIVE_SECONDS_TO_WAIT = 5;
     private static final Logger logger = LoggerFactory.getLogger(ApplicationStartupDependentResourceChecker.class);
 
     private final ApplicationStartupDependentResource applicationStartupDependentResource;
+    private final Consumer<Duration> waiter;
 
-    public ApplicationStartupDependentResourceChecker(ApplicationStartupDependentResource applicationStartupDependentResource) {
+    ApplicationStartupDependentResourceChecker(ApplicationStartupDependentResource applicationStartupDependentResource, Consumer<Duration> waiter) {
```

Comment:
If this is called in the `run` method, so once at startup, any reason we have to pass the waiter in instead of instantiating it directly in this class?

Reply:
Mocking in tests